borg/key.py

```diff
@@ -205,7 +206,20 @@ def env_passphrase(cls, default=None):
 
     @classmethod
     def getpass(cls, prompt):
-        return cls(getpass.getpass(prompt))
+        askpass = os.environ.get('BORG_ASKPASS', None)
+        if askpass is None:
+            return cls(getpass.getpass(prompt))
+        else:
+            try:
+                return cls(
+                    check_output([askpass, prompt])
+                        .decode(sys.getfilesystemencoding(), 'surrogateescape')
+                        .strip()
+                )
+            except CalledProcessError:
+                # This follows what openssl does with non-zero exit codes on SSH_ASKPASS
+                #https://github.com/openssh/openssh-portable/blob/94141b7ade24afceeb6762a3f99e09e47a6c42b6/readpass.c#L152
+                return ""
```

Comment:
hmm guess this is ok for create/extract - if there are issues, you'll get an empty password and it will fail to decrypt your key.

but for init - will it protect your key with an empty passphrase without you realizing it?